_pytest/assertion/util.py

```diff
@@ -225,10 +225,18 @@ def _compare_eq_iterable(left, right, verbose=False):
     # dynamic import to speedup pytest
     import difflib
 
-    left = pprint.pformat(left).splitlines()
-    right = pprint.pformat(right).splitlines()
-    explanation = [u('Full diff:')]
-    explanation.extend(line.strip() for line in difflib.ndiff(left, right))
+    try:
+        left_formatting = pprint.pformat(left).splitlines()
+        right_formatting = pprint.pformat(right).splitlines()
+        explanation = [u('Full diff:')]
+    except RuntimeError:
+        # hack: PrettyPrinter.pformat() in python 2 fails when formatting items that can't be sorted(), ie, calling
+        # sorted() on a list would raise. See issue #718.
+        # As a workaround, the full diff is generated by using the repr() string of each item of each container.
+        left_formatting = sorted(map(repr, left))
+        right_formatting = sorted(map(repr, right))
+        explanation = [u('Full diff (fallback to calling repr on each item):')]
```

Comment:
Perhaps it would be nice to add some explanation as to why it is falling back to calling repr?

"Full diff (failed to print-format, fallback to repr() on each item):'"